packages/client/src/links/core.ts

```diff
@@ -58,8 +65,17 @@ export type TRPCFetch = (
 
 export type LinkRuntimeOptions = Readonly<{
   transformer: DataTransformer;
+  /**
+   * @deprecated use `headers` on your `httpLink` / `httpBatchLink` instead
```

Comment:
For most users I believe it would be inconvenient to have to define both links in order to add headers. Wouldn’t us make sense to preserve this as base headers. 🤔

Reply:
It doesn't make sense to have both places, I don't wanna support 2 APIs that do the same thing. One of the non-articulated design principles of tRPC is that it's better to be closer to the metal than to offer inflexible shortcuts.